yarn-project/foundation/src/serialize/buffer_reader.test.ts

```diff
@@ -173,4 +173,68 @@ describe('buffer reader', () => {
       expect(bufferReader.peekBytes(10)).toEqual(Buffer.from(ARRAY.slice(0, 10)));
     });
   });
+
+  describe('error handling', () => {
```

Comment:
Could you add a peek failure as well?

Reply:
Added in https://github.com/AztecProtocol/aztec-packages/pull/7149/commits/1050840bc8e7630f53159a42143d035c54c816dc